### PR TITLE
[YUNIKORN-2460] Unify ask and allocation APIs

### PR DESCRIPTION
**Pull request by craigcondit**
### What is this PR for?
Remove ask-specific APIs and use UpdateAllocation for all communication.

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [x] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2460

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: scheduler-interface-spec.md [hunk 1]
```diff
@@ -584,49 +535,37 @@ message AllocationRelease {
...
++-----------+ +--------+ +-------+
+|SCHEDULABLE|+-------->|DRAINING|+---------->|REMOVED|
++-----------+ +--------+ +-------+
+^ Asked by + Aasked by
+| RM to DRAIN | RM to REMOVE
+| |
++---------------------+
+Asked by RM to
+SCHEDULE again
+
```
Comment (wilfred-s) [inline scheduler-interface-spec.md:561]: revert to old formatting

**Review by wilfred-s — CHANGES_REQUESTED** [review]
formatting change only, did not see any other issues

**Comment by wilfred-s** [comment]
BTW this PR should move to the ready state. As long as we do not update the core and k8shim reference in the go.mod file it will not be picked up.

File: scheduler-interface-spec.md (hunk 1 above)
Comment (craigcondit) [reply scheduler-interface-spec.md:561]: The formatting is changed because the old two graphs are combined into one new one.

**Comment by wilfred-s** [comment]
> The formatting is changed because the old two graphs are combined into one new one.

Filed [YUNIKORN-2809](https://issues.apache.org/jira/browse/YUNIKORN-2809) to fix the formatting of the node state transition as it currently does not display as it should

**Pull request closed without merging**